tests/testing_four_c_simulation.py

```diff
@@ -592,32 +624,110 @@ def test_four_c_simulation_rotated_beam_axis(self):
         self.run_four_c_test("rotated_beam_axis", input_file)
         self.run_four_c_test("rotated_beam_axis", input_file, nox_xml_file="xml_name")
 
-    def test_four_c_simulation_dirichlet_boundary_to_neumann_boundary(self):
+    def test_four_c_simulation_dbc_monitor_to_input(self):
```

Comment:
I think I now understand the added test cases, so sorry if my previous comment caused some confusion. We should unify `test_four_c_simulation_dbc_monitor_to_input` and `test_four_c_simulation_all_dbc_monitor_values_to_input` as they produce exactly the same files. So basically we should create the following structure:
```python
    def xtest_four_c_simulation_dbc_monitor_to_input_linear(self, *, type=None):
        ...
        if type == "dbc_monitor_to_input":
            dbc_monitor_to_input(
                restart_simulation,
                os.path.join(
                    testing_temp,
                    initial_run_name,
                    f"{initial_run_name}_monitor_dbc",
                    f"{initial_run_name}_102_monitor_dbc.csv",
                ),
                n_dof=9,
                function=function_nbc,
            )
        elif type == "dbc_monitor_to_input_all":
            all_dbc_monitor_values_to_input(
                restart_simulation,
                os.path.join(
                    testing_temp,
                    initial_run_name,
                    f"{initial_run_name}_monitor_dbc",
                    f"{initial_run_name}_102_monitor_dbc.csv",
                ),
                n_dof=9,
                time_span=[10 * 0.5, 21 * 0.5],
                fun_array=[function_nbc, function_nbc, function_nbc],
            )
        ...

    def test_four_c_simulation_dbc_monitor_to_input_linear(self):
        test_four_c_simulation_dbc_monitor_to_input_linear(self, type="dbc_monitor_to_input")

    def test_four_c_simulation_dbc_monitor_to_input_all_linear(self):
        test_four_c_simulation_dbc_monitor_to_input_linear(self, type="dbc_monitor_to_input_all")
```
This should save a lot of duplicated code.

Reply:
Yes, this is now better to understandable. I just used the name of the folder(`initial_run_name`) to distinguish between them.